Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py

```diff
@@ -0,0 +1,406 @@
+            iocs.append(indicator_dict)
+        elif itype == 'software':
+            indicator_dict = {'Type': 'Software', 'Value': indicator}
+
+    demisto.info("Length before IOCS are uploaded to Intsights...")
+    demisto.results(len(iocs))
+
+    # Connect to Intsights
+    # Prepare POST request header
+    code = base64.b64encode(bytes("{}:{}".format(intsights_id, intsights_key), 'utf-8'))
+    code = code.decode("utf-8")
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': 'Basic {}'.format(code),
+        'Zone': 'us'
+    }
+
+    # Template of IOCs dictionary to upoad to Intsights
+    custom_ioc_source = {
+        'DocumentDetails': {
+            'Name': doc_source_name,
+            'Description': 'IOCs imported from Taxii server via Demisto',
+            'ConfidenceLevel': ConfidenceLevel
+        },
+        'Iocs': [{}]
+    }
+
+    # Add downloaded IOCs to dictonary
+    custom_ioc_source['Iocs'] = iocs
+
+    # Delete existing source (IOCs) in Intsights
+    demisto.info("Deleting IOC source document if it exists...")
+    try:
+        response = requests.get(intsights_api + intsights_get_sources, headers=headers)
+
+        if response:
+            response.raise_for_status()
+            for source in response.json()['Files']:
+
+                if source['Name'] == doc_source_name:
+                    demisto.info('Deleting existing document in Intsights...')
+                    response = requests.delete(
+                        '{}{}/{}'.format(intsights_api, intsights_del_source, source['_id']),
+                        data=json.dumps(custom_ioc_source),
+                        headers=headers
+                    )
+    except Exception as e:
+        sendError('Unable to delete a source - {}'.format(e))
+
+    # Add IOCs to Intsights
+    demisto.info('Adding IOC source document to Intsights...')
+    try:
+        response = requests.post(intsights_api + intsights_add_source, data=json.dumps(custom_ioc_source), headers=headers)
+        if response:
+            response.raise_for_status()
+            demisto.info('IOCs document was successfully added')
+
+            title = '{} - {} document created'.format(INTEGRATION_NAME, doc_source_name)
+
+            ioc_count = len(iocs)
+
+            context = {
+                INTEGRATION_NAME + '.ioc_count': ioc_count,
+            }
+
+            tableHeaders = ['IOC count']  # type: List[str]
+            human_readable = tableToMarkdown(title, str(ioc_count), tableHeaders)
+            return human_readable, context, ioc_count
+
+        else:
+            sendError('Unable to add IOCs document - {}'.format(response.content))
+    except Exception as e:
+        sendError('Unable to add IOCs document - {}'.format(e))
+
+
+# The command demisto.command() holds the command sent from the user.
+if demisto.command() == 'trustar-get-IOC-data':
+    uncleanListOfIndicators = get_ioc_data()
```

Comment:
Please note that `get_ioc_data ` might fail . As commands must raise exceptions on failure, you should catch those exceptions and use `return_error ` in order to handle it correctly.